File: apzkr-pzpi-21-11-milennyi-ivan/Task3-WebClient/src/viewModels/FeedingLogVM.ts
import { timestampToDate } from "../utils/helpers";

// Клас для відображення записів годування
export class FeedingLogVM {
    id: number; // Ідентифікатор запису годування
    date: string; // Дата годування
    amount: number; // Кількість корму в кг
    shepherdName: string | null; // Ім'я пастуха
    shepherdSurname: string | null; // Прізвище пастуха
    sheepId: number; // Ідентифікатор вівці

    constructor(id: number, timestamp: number, amount: number, sheepId: number, shepherdName: string | null, shepherdSurname: string | null) {
        this.id = id;
        this.amount = amount / 1000;
        this.shepherdName = shepherdName;
        this.shepherdSurname = shepherdSurname;
        this.sheepId = sheepId;
        this.date = timestampToDate(timestamp);
    }
}

File: apzkr-pzpi-21-11-milennyi-ivan/Task1-Server/src/view_models/feeding_log_vm.rs
use serde::Serialize;
use sqlx::FromRow;
use utoipa::ToSchema;

// Структура, що представляє представлення логу годування (ViewModel)
#[derive(Debug, Serialize, FromRow, ToSchema)]
pub(crate) struct FeedingLogVM {
    id: u64, // Ідентифікатор логу годування
    timestamp: u64, // Час годування (у вигляді таймстемпу)
    amount: u64, // Кількість корму
    shepherd_name: Option<String>, // Ім'я пастуха (може бути відсутнім)
    shepherd_surname: Option<String>, // Прізвище пастуха (може бути відсутнім)
    sheep_id: u64 // Ідентифікатор вівці
}
